src/main/java/org/minions/devfund/bruno/html/HtmlTag.java

```diff
@@ -0,0 +1,161 @@
+            "xml", "area", "base", "basefont", "br", "col", "frame", "hr", "img", "input", "link", "meta", "param"));
+    protected static final String WHITESPACE = " \f\n\r\t";
+    protected final String element;
+    protected final boolean openTag;
+
+    /**
+     * Initializes an instance of {@link HtmlTag}.
+     *
+     * @param element   element name.
+     * @param isOpenTag boolean value.
+     */
+    public HtmlTag(final String element, final boolean isOpenTag) {
+        this.element = element.toLowerCase();
+        openTag = isOpenTag;
+    }
+
+    /**
+     * Checks whether this is the opening tag.
+     *
+     * @return false if is open the tag.
+     */
+    public boolean isOpenTag() {
+        return openTag && !isSelfClosing();
+    }
+
+    /**
+     * Checks whether an HtmlTag other is the matching open/close tag to itself.
+     *
+     * @param other Htmltag object.
+     * @return boolean.
+     */
+    public boolean matches(final HtmlTag other) {
+        return other != null
+                && element.equalsIgnoreCase(other.element)
+                && openTag != other.openTag;
+    }
+
+    /**
+     * Checks whether a tag is self-closing.
+     *
+     * @return if is self closing return true.
+     */
+    public boolean isSelfClosing() {
+        return SELF_CLOSING_TAGS.contains(element);
+    }
+
+    /**
+     * Checks if the Html tag object are equals.
+     *
+     * @param obj Html tag Object.
+     * @return a boolean value.
+     */
+    @Override
+    public boolean equals(final Object obj) {
+        if (obj instanceof HtmlTag) {
+            HtmlTag other = (HtmlTag) obj;
+            return element.equals(other.element)
+                    && openTag == other.openTag;
+        }
+        return false;
+    }
+
+    /**
+     * Overrides hash code.
+     *
+     * @return element.
+     */
+    @Override
+    public int hashCode() {
+        return element.hashCode();
+    }
+
+    /**
+     * Builds tag according the element.
+     *
+     * @return a string.
+     */
+    public String toString() {
+        return "<".concat(openTag ? "" : "/")
+                .concat(element.equals("!--") ? "!-- --" : element).concat(">");
+    }
+
+    /**
+     * Checks the text parameter is a tag and put in queue.
+     *
+     * @param text text from html file.
+     * @return a queue.
+     */
+    public static Queue<HtmlTag> tokenize(final String text) {
+        StringBuffer buf = new StringBuffer(text);
+        Queue<HtmlTag> queue = new LinkedList<>();
+        HtmlTag nextTag = nextTag(buf);
+        while (nextTag != null) {
+            queue.add(nextTag);
+            nextTag = nextTag(buf);
+        }
+        return queue;
+    }
+
+    /**
+     * Checks the buf parameter.
+     *
+     * @param buf text from html file.
+     * @return a html tag.
+     */
+    protected static HtmlTag nextTag(final StringBuffer buf) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace the synchronized class "StringBuffer" by an unsynchronized one such as "StringBuilder". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1149)